File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    answer = 1
    dic = {}
    for i in clothes:
        if i[1] not in dic:
            dic[i[1]] = 1
        else:
            dic[i[1]] +=1
    for i in dic:
        answer *= (dic[i] + 1)
    return answer - 1

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;
class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        HashMap<String,Integer> map = new HashMap<>();
        for (int i=0;i<clothes.length;i++) {
            if (map.containsKey(clothes[i][1])) {
                map.put(clothes[i][1],map.get(clothes[i][1])+1);
            }else {
                map.put(clothes[i][1],1);
            }
        }
        for (String i : map.keySet()) {
            answer *= (map.get(i)+1); 
        }
        return answer-1;
    }
}

    